### PR TITLE
feat: add mnemonic import view for stardust mobile

### LINKED ISSUE #4818: [Task]: Add mnemonic phrase import for onboarding
### Task description

To recover a profile with mnemonics while onboarding the ProfileRecoveryRouter as well as ImportMnemonicPhraseView and SuccessView have to be added and adjusted for mobile usage.

### Requirements

- ProfileRecoveryRouter implemented and adjusted
- ImportMnemonicPhrasView implemented and adjusted
- SuccessView implemented and adjusted

### Creation checklist

- [X] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by paul-boegelsack**
## Summary

This PR adds the router component to restore a profile via mnemonics.

## Changelog

```
- Adds router components for mnemonic recovery
- Updates profile setup for recovery
- Enables mnemonic recovery feature for stardust dev net
```

## Relevant Issues

Closes #4818

## Testing

Until the mobile bindings are ready, the `dev-mode/mobile-stardust` branch has to be utilized for testing:

1 `git checkout dev-mode/mobile-stardust` 
2 `git merge feat-mobile/new-mnemonic-import-view`
3 `yarn (root directory)`
4 `yarn build (packages/desktop)`
5 `yarn start (packages/desktop)`

press `ctrl + c` to change the color theme
press `ctrl + k` to toggle the keyboard

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
  - [ ] MacOS
  - [ ] Linux
  - [ ] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

...

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/mobile/routes/onboarding/views/profile-recovery/ProfileRecoveryRouter.svelte [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    import { localize } from '@core/i18n'
+    import {
+        profileRecoveryRoute,
+        profileRecoveryRouter,
+        ProfileRecoveryRouter,
+        ProfileRecoveryRoute,
+    } from '@core/router'
+    import { showAppNotification } from '@lib/notifications'
+
+    setContext<ProfileRecoveryRouter>('importRouter', $profileRecoveryRouter)
```
Comment (begonaalvarezd) [inline packages/mobile/routes/onboarding/views/profile-recovery/ProfileRecoveryRouter.svelte:13]: I dont think this context is used, looks like debris

File: packages/mobile/routes/onboarding/views/profile-recovery/ProfileRecoveryRouter.svelte [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    $profileRecoveryRouter.resetRoute()
```
Comment (begonaalvarezd) [inline packages/mobile/routes/onboarding/views/profile-recovery/ProfileRecoveryRouter.svelte:14]: I personally prefer doing this inside onMount so its more clear when its being triggered

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]
Looking amazing 🚀 just added 2 mini comments to cleanup a bit the code but this is almost ready to merge, great work 💪🏼

**Comment by paul-boegelsack** [comment]
> Looking amazing 🚀 just added 2 mini comments to cleanup a bit the code but this is almost ready to merge, great work 💪🏼

Alright, thank you! 👍

**Review by begonaalvarezd — APPROVED** [review]
🔥 on fire!

**Pull request merged**